### PR TITLE
Silence the at-pure deprecation warning.

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Review by aviatesk — APPROVED** [review]

**Comment by aviatesk** [comment]
Thanks.

**Comment by fredrikekre** [comment]
Pkg tests are failing now since something is running with `--depwarn=error`.

**Comment by DilumAluthge** [comment]
> Pkg tests are failing now since something is running with `--depwarn=error`.

https://github.com/JuliaData/Tables.jl/issues/327

**Comment by DilumAluthge** [comment]
@aviatesk What is the earliest version of Julia for which `Base.@assume_effects :foldable ex` is available?

**Comment by KristofferC** [comment]
What's the gain in having a deprecation of this at all? Updating to the new spelling is just a name change that has zero productive value. Just let this silently sit in `deprecated.jl` and if people are so inclined they can change it to the new spelling.

**Comment by DilumAluthge** [comment]
Seems fine to leave the note in the documentation?

**Comment by fredrikekre** [comment]
There is already a note.

File: base/deprecated.jl [hunk 1]
```diff
@@ -378,14 +378,8 @@ helping for type inference.
     In Julia 1.8 and higher, it is favorable to use [`@assume_effects`](@ref) instead of `@pure`.
     This is because `@assume_effects` allows a finer grained control over Julia's purity
     modeling and the effect system enables a wider range of optimizations.
-
-!!! note
```
Comment (KristofferC) [inline base/deprecated.jl:382]: We can still have the note saying to prefer the new style maybe?

Comment (fredrikekre) [reply base/deprecated.jl:382]: Something more than the sentence just above?

Comment (KristofferC) [reply base/deprecated.jl:382]: Yeah you're right, it basically says that there.

**Pull request merged**